pkg/types/aws/platform.go

```diff
@@ -144,6 +149,92 @@ type ServiceEndpoint struct {
 	URL string `json:"url"`
 }
 
+// VPCSpec configures the VPC for the cluster.
+type VPCSpec struct {
+	// Subnets defines the subnets in an existing VPC and can optionally specify their intended roles.
+	// If no roles are specified on any subnet, then the subnet roles are decided automatically.
+	// In this case, the VPC must not contain any subnets without the kubernetes.io/cluster/<cluster-id> tag.
+	//
+	// For manually specified subnet role selection, each subnet must have at least one assigned role,
+	// and the ClusterNode, IngressControllerLB, ControlPlaneExternalLB, and ControlPlaneInternalLB roles must be assigned to at least one subnet.
+	// However, if the cluster scope is internal, then ControlPlaneExternalLB is not required.
+	//
+	// Subnets must contain unique IDs, must not exceed a total of 40 subnets,
+	// and can include no more than 10 subnets with the IngressController role.
+	//
+	// Leave this field unset to have the installer create subnets in a new VPC on your behalf.
+	//
+	// Subnets is available in TechPreview.
+	//
+	// +kubebuilder:validation:MaxItems=50
+	// +listType=atomic
```

Comment:
nit: What do we think about making this a map with the `key=ID`

Reply:
This makes sense to me. I am just wondering if there is any user-facing intention behind using `atomic` here (i.e. like each sets of subnets need to be specified together) - I might be overthinking again?

Reply:
the current slice of `Subnet`s implementation makes sense. `ID` is a required field today, but in the future we may extend the Subnets type to allow greater functionality. For example, we may make `ID` optional and add an availability zone field to allow specifying more details for IPI-provisioned subnets.

Reply:
Right, thank you! Then I will keep it as it is defined now (i.e. as in the proposal).